src/main/java/com/limechain/beefy/state/BeefyState.java

```diff
@@ -207,4 +215,22 @@ private void persistJustification(BigInteger blockNumber, SignedCommitment justi
                 justification
         );
     }
+
+    private Commitment getCommitment(BigInteger blockNumber, BigInteger setId) {
+        BlockHeader blockHeader = blockState.getHeaderByNumber(blockNumber);
+        byte[] mmrHash = extractMmrRootHash(blockHeader);
+        PayloadElement payloadElement = new PayloadElement(BeefyPayloadId.MMR, mmrHash);
+
+        return new Commitment(Collections.singletonList(payloadElement), blockNumber, setId);
+    }
+
+    private byte[] extractMmrRootHash(BlockHeader blockHeader) {
+        return DigestHelper.getBeefyConsensusMessages(blockHeader.getDigest())
+                .stream().map(BeefyConsensusMessage::getMmrRootHash)
+                .filter(Objects::nonNull)
+                .findFirst()
+                .orElseThrow(() -> new BeefyGenericException(
+                        String.format("No MMR digest found in block header: %d", blockHeader.getBlockNumber()))
+                );
+    }
```

Comment:
```suggestion
        return DigestHelper.getBeefyConsensusMessages(blockHeader.getDigest()).stream()
                .map(BeefyConsensusMessage::getMmrRootHash)
                .findFirst()
                .orElseThrow(() -> new BeefyGenericException(
                        String.format("No MMR digest found in block header: %d", blockHeader.getBlockNumber()))
                );
    }
```

`DigestHelper` won't return null values.